libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/api/MapboxOffboardSpeechPlayer.kt

```diff
@@ -0,0 +1,120 @@
+package com.mapbox.navigation.ui.voice.api
+
+import android.content.Context
+import android.media.MediaPlayer
+import com.mapbox.navigation.ui.base.api.voice.SpeechPlayer
+import com.mapbox.navigation.ui.base.api.voice.SpeechPlayerCallback
+import com.mapbox.navigation.ui.base.model.voice.Announcement
+import com.mapbox.navigation.ui.base.model.voice.SpeechState
+import java.io.File
+import java.io.FileInputStream
+import java.io.FileNotFoundException
+import java.io.IOException
+
+/**
+ * Online implementation of [SpeechPlayer].
+ * Will retrieve synthesized speech mp3s from Mapbox's API Voice.
+ * @property context Context
+ * @property accessToken String
+ * @property language String
+ */
+internal class MapboxOffboardSpeechPlayer(
+    private val context: Context,
+    private val accessToken: String,
+    private val language: String
+) : SpeechPlayer {
+
+    private var mediaPlayer: MediaPlayer? = null
+    private var volumeLevel: Float = DEFAULT_VOLUME_LEVEL
+    private lateinit var clientCallback: SpeechPlayerCallback
+    private var currentPlay: SpeechState.Play = SpeechState.Play(Announcement("", null, null))
+
+    /**
+     * Given [SpeechState.Play] [Announcement] the method will play the voice instruction.
+     * If a voice instruction is already playing or other announcement are already queued,
+     * the given voice instruction will be queued to play after.
+     * @param state SpeechState Play Announcement object including the announcement text
+     * and optionally a synthesized speech mp3.
+     * @param callback
+     */
+    override fun play(state: SpeechState.Play, callback: SpeechPlayerCallback) {
+        clientCallback = callback
+        currentPlay = state
```

Comment:
How about we assert here that `play` is never invoked when the previous announcement has not finished playing? This way we'll find bugs much quicker.